qiskit/extensions/standard/ccx.py

```diff
@@ -77,14 +78,24 @@ def reapply(self, circ):
 
 def ccx(self, ctl1, ctl2, tgt):
     """Apply Toffoli to from ctl1 and ctl2 to tgt."""
-    if isinstance(ctl1, QuantumRegister) and \
-       isinstance(ctl2, QuantumRegister) and \
-       isinstance(tgt, QuantumRegister) and \
-       len(ctl1) == len(tgt) and len(ctl2) == len(tgt):
-        instructions = InstructionSet()
-        for i in range(ctl1.size):
-            instructions.add(self.ccx((ctl1, i), (ctl2, i), (tgt, i)))
-        return instructions
+    if isinstance(ctl1, QuantumRegister):
+        ctl1 = [(ctl1, i) for i in range(len(ctl1))]
+    if isinstance(ctl2, QuantumRegister):
+        ctl2 = [(ctl2, i) for i in range(len(ctl2))]
+    if isinstance(tgt, QuantumRegister):
+        tgt = [(tgt, i) for i in range(len(tgt))]
+
+    if ctl1 and ctl2 and tgt:
```

Comment:
I once mistakenly typed "qc" instead of "qr" in a call to `ccx`, passing a QuantumCircuit slice instead of a QuantumRegister slice, and no error message was thrown. This an attempt to catch passing empty slices.